provisioning/src/main/java/org/wildfly/build/util/POSIXFinePrivilegesHelper.java

```diff
@@ -0,0 +1,46 @@
+package org.wildfly.build.util;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.attribute.PosixFilePermission;
+import java.nio.file.attribute.PosixFilePermissions;
+import java.util.Set;
+
+import static org.wildfly.build.provisioning.ServerProvisioner.OS_WINDOWS;
+
+public class POSIXFinePrivilegesHelper {
+
+    // TODO: ugly
+    private static final Path NO_PATH_PRODIVED = null;
+
+    private static final String JAR_FILE_MASK = "rw-r--r--";
+    private static final String PROPERTIES_FILE_MASK = "rw-------";
+
+    public static void fineTunePrivileges(Path path) throws IOException {
+        if (!OS_WINDOWS) Files.setPosixFilePermissions(path,getPermissionsMasks(NO_PATH_PRODIVED,path));
+    }
+
+    private static Set<PosixFilePermission> getPermissionsMasks(Path src, Path dest) {
+        final String suffix = dest.toAbsolutePath().toString();
+        if ( suffix.endsWith(".jar") ) {
+            System.out.println("JAR:" + dest);
+            return PosixFilePermissions.fromString(JAR_FILE_MASK);
+        } else if ( suffix.endsWith(".properties") ) {
+            System.out.println("PROPS:" + dest);
```

Comment:
another

Reply:
darn, sorry, I forgot to remove all of those sysout. My bad. However, it is just a proposal, it's not finished, as some files are not yet covered. At this stage, I want to know if we agree to fix this issue here (wildlly-build-tool).